client/src/state.rs

```diff
@@ -0,0 +1,307 @@
+use std::sync::Arc;
+
+use celestia_proto::cosmos::bank::v1beta1::MsgSend;
+use celestia_proto::cosmos::staking::v1beta1::{
+    MsgBeginRedelegate, MsgCancelUnbondingDelegation, MsgDelegate, MsgUndelegate,
+};
+use celestia_rpc::StateClient;
+use celestia_types::state::{
+    AccAddress, Address, Coin, PageRequest, QueryDelegationResponse, QueryRedelegationsResponse,
+    QueryUnbondingDelegationResponse, ValAddress,
+};
+use celestia_types::Blob;
+use k256::ecdsa::VerifyingKey;
+
+use crate::client::Context;
+use crate::tx::{IntoAny, TxConfig, TxInfo};
+use crate::utils::height_i64;
+use crate::Result;
+
+/// State API for quering and submiting TXs to a consensus node.
+pub struct StateApi {
+    ctx: Arc<Context>,
+}
+
+impl StateApi {
+    pub(crate) fn new(ctx: Arc<Context>) -> StateApi {
+        StateApi { ctx }
+    }
+
+    /// Returns the public key of the signer.
+    pub fn pubkey(&self) -> Result<VerifyingKey> {
+        self.ctx.pubkey().cloned()
+    }
+
+    /// Returns the address of signer.
+    pub fn account_address(&self) -> Result<AccAddress> {
+        let pubkey = self.ctx.pubkey()?.to_owned();
+        Ok(AccAddress::new(pubkey.into()))
+    }
```

Comment:
Should I move these two in `Client`? Go celestia-node has only `AccountAddress` and under `State.API`.

Reply:
I think I like these here, they seem to make sense together with the other ones here.

Reply:
moved in Client